File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        for i in range(len(nums)):
            print("I number is: ", nums[i])
            for j in range(i + 1, len(nums)):
                print("J number is: ", nums[j])
                if nums[i] == nums[j]:
                    return True 
        
        return False 

File: Data Structures & Algorithms/contains-duplicate-ii/submission-1.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        // fixed sliding window problem. 
        HashSet<Integer> set = new HashSet<>(); 

        int L = 0; 
        for (int R = 0; R < nums.length; R++)
        {
            if (R - L > k)
            {
                set.remove(nums[L]);
                L++;
            }
            if (set.contains(nums[R])) return true;
            set.add(nums[R]);
        }
        return false;
    }
}